### PR TITLE
Sync v2-5-stable with v2-5-test to release 2.5.3

### PR DESCRIPTION
**Pull request by pierrejeambrun**
`2.5.3rc2` :tada:

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 .. towncrier release notes start
 
-Airflow 2.5.3 (2023-03-30)
+Airflow 2.5.3 (2023-03-31)
```
Comment (eladkal) [inline RELEASE_NOTES.rst:24]: I think we should add the fix PR to the change log.. it's a small release so if we can list all relevant PRs it's better

Comment (pierrejeambrun) [reply RELEASE_NOTES.rst:24]: It fixes a bug that wasn't released yet https://github.com/apache/airflow/pull/29971. (part of 2.5.3).

What entry would we like for  it ?  (https://github.com/apache/airflow/pull/30355)

Comment (pierrejeambrun) [reply RELEASE_NOTES.rst:24]: Done, thanks for the additional context provided on slack :)

**Review by eladkal — APPROVED** [review]

**Comment by pierrejeambrun** [comment]
https://github.com/apache/airflow/pull/30355 messed up a little bit the groups collapse/expend feature. We are waiting for a last adjustment before moving forward with rc2.

cc: @bbovenzi

**Comment by bbovenzi** [comment]
Just wanted to have the groups zoom out slightly. Didn't realize how bad it got with multi-nested groups:https://github.com/apache/airflow/pull/30360

**Review by ephraimbuddy — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**